### ISSUE TITLE
Feature request: focus specific frame

### ISSUE THREAD

**Issue by cbf305** [herbstluftwm/herbstluftwm#1164]
The more I tweak HLWM to better my workflow, the more I find that at times it would be nice to be able to focus a specific frame either with a keybinding or in a shell script.  I was thinking something like ```herbstclient focus_frame <tag> <frame_index>```, where the tag could be the name, number or current/focused if omitted and the frame_index would follow the "Frame Index" section of the documentation and/or maybe the frame index stored in ```tags.<by-name | number | focus>.tiling.focused_frame.index```. 

So if I wanted to spawn a new client in the first empty frame, rather then having a script cycle and check the current focused frame to see if it's empty before spawning a new client, I could do something like ```herbstclient chain - focus_frame e - spawn "$1"```, which would focus the first empty frame on the focused tag and spawn the app passed in.

I dunno if I am alone on this one, but I think there are many possibilities for having this functionality.

Thanks!

**Comment by t-wissmann** [comment]
You can accomplish the same thing in a cleaner way already now. The  `chain` command you gave has an issue: you need to wait until the application appears (`spawn` only spawns a process, it may take some time until the window appears).

With cleaner I mean the following:
```
herbstclient chain - rule once index=e focus=on - spawn "$1"
```
This will add a one-time rule (`once`) that moves the next window that shows up to an empty frame and focuses it.

**Comment by cbf305** [comment]
Interesting, I never thought about looking at this from a rule perspective.  This is perfect and does exactly what I was looking for.

Thanks!

**Comment by Pinoinha** [comment]
Hello! I'd like to expand a little bit on this issue.

While the rule works great for focusing the empty frame, I've been having some problems with a script I'm trying to write.

The intended functionality for the script is as follows:
1) If the currently focused frame is empty, open a terminal on it;
2) Else, if it's not empty, then split it vertically and open a terminal on the new empty frame

What I've been able to pull up is the following command:

```
herbstclient chain , split right 0.5 , rule once focus=on index=e , spawn $TERMINAL 
```

But obviously this only works for the second use case I've stated above, since the new frame will always be created.

I know I'm missing something, so any advice is highly appreciated.
